shared-bindings/framebufferio/FramebufferDisplay.c

```diff
@@ -0,0 +1,452 @@
+    .proxy = {(mp_obj_t)&framebufferio_framebufferdisplay_get_width_obj,
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+//|   .. attribute:: height
+//|
+//|	Gets the height of the framebuffer
+//|
+STATIC mp_obj_t framebufferio_framebufferdisplay_obj_get_height(mp_obj_t self_in) {
+    framebufferio_framebufferdisplay_obj_t *self = native_display(self_in);
+    return MP_OBJ_NEW_SMALL_INT(common_hal_framebufferio_framebufferdisplay_get_height(self));
+}
+MP_DEFINE_CONST_FUN_OBJ_1(framebufferio_framebufferdisplay_get_height_obj, framebufferio_framebufferdisplay_obj_get_height);
+
+const mp_obj_property_t framebufferio_framebufferdisplay_height_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&framebufferio_framebufferdisplay_get_height_obj,
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+//|   .. attribute:: rotation
+//|
+//|     The rotation of the display as an int in degrees.
+//|
+STATIC mp_obj_t framebufferio_framebufferdisplay_obj_get_rotation(mp_obj_t self_in) {
+    framebufferio_framebufferdisplay_obj_t *self = native_display(self_in);
+    return MP_OBJ_NEW_SMALL_INT(common_hal_framebufferio_framebufferdisplay_get_rotation(self));
+}
+MP_DEFINE_CONST_FUN_OBJ_1(framebufferio_framebufferdisplay_get_rotation_obj, framebufferio_framebufferdisplay_obj_get_rotation);
+STATIC mp_obj_t framebufferio_framebufferdisplay_obj_set_rotation(mp_obj_t self_in, mp_obj_t value) {
+    framebufferio_framebufferdisplay_obj_t *self = native_display(self_in);
+    common_hal_framebufferio_framebufferdisplay_set_rotation(self, mp_obj_get_int(value));
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_2(framebufferio_framebufferdisplay_set_rotation_obj, framebufferio_framebufferdisplay_obj_set_rotation);
+
+
+const mp_obj_property_t framebufferio_framebufferdisplay_rotation_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&framebufferio_framebufferdisplay_get_rotation_obj,
+              (mp_obj_t)&framebufferio_framebufferdisplay_set_rotation_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+//|   .. attribute:: framebuffer
+//|
+//|	The framebuffer being used by the display
+//|
+//|
+STATIC mp_obj_t framebufferio_framebufferdisplay_obj_get_framebuffer(mp_obj_t self_in) {
+    framebufferio_framebufferdisplay_obj_t *self = native_display(self_in);
+    return common_hal_framebufferio_framebufferdisplay_get_framebuffer(self);
+}
+MP_DEFINE_CONST_FUN_OBJ_1(framebufferio_framebufferdisplay_get_framebuffer_obj, framebufferio_framebufferdisplay_obj_get_framebuffer);
+
+const mp_obj_property_t framebufferio_framebufferframebuffer_obj = {
+    .base.type = &mp_type_property,
+    .proxy = {(mp_obj_t)&framebufferio_framebufferdisplay_get_framebuffer_obj,
+              (mp_obj_t)&mp_const_none_obj,
+              (mp_obj_t)&mp_const_none_obj},
+};
+
+
+//|   .. method:: fill_row(y, buffer)
```

Comment:
Why have this when one can just read the framebuffer?

Reply:
Real reason: copypasta from Display
Possible justification: this is guaranteed to be in RGB888 format, while the framebuffer is in some internal format and might not be readable from raw python code (if it's a real hardware framebuffer like imx anyway)

Reply:
kk